MergerLogic/Batching/Tile.cs

```diff
@@ -41,25 +45,35 @@ public Tile(int z, int x, int y, byte[] data)
             this.X = x;
             this.Y = y;
             this.Format = ImageFormatter.GetTileFormat(data) ?? throw new ValidationException($"Cannot create tile {this}, data is in invalid format");
+            var info = new MagickImageInfo(data);
+            this.Width = info.Width;
+            this.Height = info.Height;
             this._data = data;
-        }
 
-        public Tile(Coord cords, byte[] data)
-        {
-            this.Z = cords.Z;
-            this.X = cords.X;
-            this.Y = cords.Y;
-            this.Format = ImageFormatter.GetTileFormat(data) ?? throw new ValidationException($"Cannot create tile {this}, data is in invalid format");
-            this._data = data;
+            int allowedPixelSize = 256;
+            if (this.Width != allowedPixelSize || this.Height != allowedPixelSize)
+            {
+                throw new ArgumentException($"The image dimensions ({this.Width}x{this.Height}) does not match the allowed size ({allowedPixelSize})");
+            }
         }
 
+        public Tile(Coord cords, byte[] data) : this(cords.Z, cords.X, cords.Y, data) { }
+
         public Tile(Coord cords, IMagickImage<byte> image)
         {
             this.Z = cords.Z;
             this.X = cords.X;
             this.Y = cords.Y;
             this.Format = ImageFormatter.GetTileFormat(image) ?? throw new ValidationException($"Cannot create tile {this}, data is in invalid format");
+            this.Width = image.Width;
+            this.Height = image.Height;
             this._data = image.ToByteArray();
+
+            int allowedPixelSize = 256;
+            if (this.Width != allowedPixelSize || this.Height != allowedPixelSize)
```

Comment:
Consider move this validation logic to separated function and call it here to avoid duplicate it

Reply:
Moved this to a different function in ImageUtils - "ValidateTileSize"